### ISSUE TITLE
PLE1307: false positive

### ISSUE THREAD

**Issue by spaceone** [astral-sh/ruff#8406]
PLE1307: false positive:
```python
$ cat x.py
print('[%c]' % ('x',))
print('[%c]' % 'x')
$ ruff x.py
x.py:1:7: PLE1307 Format type does not match argument type
x.py:2:7: PLE1307 Format type does not match argument type
Found 2 errors.
```

**Comment by zanieb** [comment]
Thanks for the report. I dug into this a bit and the issue is that `%c` is an integer format type. Characters can be coerced to integers so they work at runtime, but Ruff's type analysis e.g. `analyze::type_inference::PythonType` does not support characters. I'm not sure how best to represent objects that can be both characters and strings in our type system so there's not a quick fix for this although it does seem easy enough to detect strings with a single item and special case them.
